labonneboite/scripts/rebuild_simplified_rome_naf_mapping.py

```diff
@@ -0,0 +1,200 @@
+# coding: utf8
+import os
+import re
+import itertools
+import logging
+import pandas as pd
+from slugify import slugify
+import numpy as np
+
+from labonneboite.conf import settings
+from labonneboite.common import mapping as mapping_util
+from labonneboite.common import hiring_type_util
+from labonneboite.common import geocoding
+from labonneboite.common.search import fetch_offices
+
+logging.basicConfig(level=logging.INFO)
+
+INPUT_FILENAME = os.path.join(os.path.dirname(os.path.realpath(__file__)), '../common/data/rome_naf_mapping.raw.csv')
+OUTPUT_FILENAME = os.path.join(os.path.dirname(os.path.realpath(__file__)), '../common/data/rome_naf_mapping.csv')
+CSV_DELIMITER = ','
+
+# There might sometimes be a few items more than the wanted maximum,
+# because for example we won't drop the last ROME of a given NAF
+# if this NAF is highly ranked for this ROME.
+# These threshold have been carefull selected so that:
+# - approx 90% of romes have less than 10 nafs
+# - approx 90% of nafs have less than 10 romes
+WANTED_MAXIMUM_ROMES_PER_NAF = 4
+WANTED_MAXIMUM_NAFS_PER_ROME = 3
+
+SHOW_DETAILED_STATS = False
+
+
+class RomeNafMapping:
+
+    def __init__(self):
+        self.df = pd.read_table(INPUT_FILENAME, sep=CSV_DELIMITER)
+        df = self.df
+
+        self.sorted_nafs_for_rome = {}
+        for rome in df.rome_id.unique():
+            self.sorted_nafs_for_rome[rome] = df[df.rome_id == rome].sort_values(
```

Comment:
`df[df.rome_id == rome]` : est-ce que c'est une subtilité de `pandas` ? J'ai du mal à voir pourquoi on accède à un élément de liste avec un booléen ?

Reply:
oui c'est l'univers magique des data frames de pandas, ça veut dire "select only the rows matching this condition"

Reply:
https://chrisalbon.com/python/data_wrangling/pandas_selecting_rows_on_conditions/